### DISCUSSION TITLE
Relationship of two same-structured graphs

### DISCUSSION THREAD

**Discussion by VicFisher** [pyg-team/pytorch_geometric#3781] [Q&A]
Hi. I have a problem about recognizing the relationship of two same-structured graphs:
**G1**  (x=[N1,F_x], edge_attr=[E1, F_e], pos1)
**G2**  (x=[N2,F_x], edge_attr=[E2, F_e], pos2)
with same node_features and edge_features but different nodes and edges number(N1 != N2, E1 !=E2).

- I want to utilize GNN to first update the features of G1 and G2 separately and then **pass the message of G1 to G2 with attention mechanism or something** to let them communicate with each other. Final give a general score to evaluate their relationship.

- I am a beginner of PyG and not that familiar with the APIs. So I'd like to ask for help about how to learning information across two different graphs and how to implement such idea. Any suggestion is welcom. Thanks a lot!!!

**Comment by rusty1s** [comment]
You might be interested in two of my works which implement an inter-message passing step between two different graph structures:
* https://github.com/rusty1s/deep-graph-matching-consensus
* https://github.com/rusty1s/himp-gnn

In general, there does not exist an API for your use-case, and you have to implement this in raw PyTorch. Given the node features `x1` and `x2` of the two graphs, you can then apply dot-product attention via:
```python
att = x2 @ x1.t()  # [N_2, N_1]

x2 = x2 + att.softmax(dim=-1) @ x1  # Add weighted node features from `x1` to `x2`
```

**Reply by VicFisher** [reply]
Thank you for your quick response. I would give it a try right now.
